azavea/temperate
# Add map modal to dashboard

## Overview

Adds a map modal to the dashboard for viewing mappable impacts associated with risks.

### Demo

<img width="962" alt="Screen Shot 2019-11-08 at 1 02 21 PM" src="https://user-images.githubusercontent.com/447977/68499779-0ee41f00-0228-11ea-9e87-ef4d460fea4d.png">
<img width="1214" alt="Screen Shot 2019-11-08 at 1 03 08 PM" src="https://user-images.githubusercontent.com/447977/68499809-2a4f2a00-0228-11ea-8a50-9317f6bced8a.png">

# Notes

- This has caused #1316 to reappear; I didn't make additional attempts to fix it because it should be more tractable once GMaps is removed.
- The map doesn't fully initialize until the user interacts to select a layer. This is due to a race condition related to the modal initialization that requires `updateMapSize()` to be called in order for the map to initialize fully. The  visual impact of this approach is that the organization marker / bounds don't show up until the user chooses a layer. Ideally, the map would initialize fully when the modal loads, but none of the places that I tried calling `updateMapSize()` managed to make that happen, so I settled for this. This should be easier to resolve once the map initialization is simplified after GMaps is removed.
- This includes the functionality changes from the initial wireframes, meaning that there is no "Risk" dropdown. However, #1340 is not implemented.

## Testing Instructions

 * Visit the Temperate Dashboard
 * On the Overview tab, you should see a new map icon next to the indicator graph icon. 
 * Some icons should be enabled, others disabled, depending on whether there are impacts associated with that risk.
 * Clicking an enabled icon should display a map that functions close to identically to the map on the map page, but the Visualization dropdown should be limited only to those layers that are related to the grouped set of risks.
 * Head to the Map page and make sure that everything behaves as before.


Closes #1306
